drivers/flash/flash_gd32.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 BrainCo Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_FLASH_FLASH_GD32_H_
+#define ZEPHYR_DRIVERS_FLASH_FLASH_GD32_H_
+
+#include <errno.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/flash.h>
+
+#define GD32_NV_FLASH_NODE	DT_INST(0, gd_gd32_nv_flash)
+#define FLASH_GD32_SIZE		DT_REG_SIZE(GD32_NV_FLASH_NODE)
+#define FLASH_GD32_ADDR		DT_REG_ADDR(GD32_NV_FLASH_NODE)
+#define FLASH_GD32_PRG_SIZE	DT_PROP(GD32_NV_FLASH_NODE, write_block_size)
+#define FLASH_GD32_TIMEOUT	DT_PROP(GD32_NV_FLASH_NODE, max_erase_time)
+
+#if (4 == FLASH_GD32_PRG_SIZE)
+typedef uint32_t flash_prg_t;
+#elif (2 == FLASH_GD32_PRG_SIZE)
+typedef uint16_t flash_prg_t;
+#elif (1 == FLASH_GD32_PRG_SIZE)
+typedef uint8_t flash_prg_t;
+#else
+#error "Invalid write-block-size value in FMC DTS"
+#endif
+
+/* Helper for conditional compilation directives, KB have a type casting. */
+#define PRE_KB(x) ((x) << 10)
```

Comment:
Use `KB` macro from `sys/util.h`.

Reply:
`KB` can not be used for preprocessor math operation, it have a type casting `(size_t)`.